modules/dhcp/providers/server/isc.rb

```diff
@@ -146,31 +146,34 @@ def loadSubnetData subnet
       initialize_memory_store_with_dhcp_records(parse_config_and_leases_for_records)
     end
 
+    SUBNET_BLOCK_REGEX = /subnet\s+([\d\.]+)\s+netmask\s+([\d\.]+)\s*\{([\w-]+\{[^{}]*\}|[\w-][^{}]*;)*\}/
```

Comment:
https://github.com/theforeman/puppet-dhcp/blob/master/templates/dhcpd.pool.erb is an example of just how messy a subnet definition can be.